pkg/cmd/historical_logs.go

```diff
@@ -107,40 +127,321 @@ func (o *LogParameters) Execute(streams genericclioptions.IOStreams) error {
+		}
 	}
 
-	responseBody, err := ioutil.ReadAll(res.Body)
+	var logList []string
+	if len(podList) > 0 {
+		for _, pod := range podList {
+
+			o.Podname = pod
+			response, err := o.makeHttpRequest(ApiUrl)
+
+			if err != nil {
+				return err
+			}
+
+			responseBody, err := ioutil.ReadAll(response.Body)
+			if err != nil {
+				return fmt.Errorf("failed to read response: %v", err)
+			}
+
+			err = getLogs(responseBody, &logList, o.Prefix)
+			if err != nil {
+				return err
+			}
+		}
+
+		err = printLogs(logList, streams, o.Limit)
+		if err != nil {
+			return err
+		}
+
+	} else {
+
+		response, err := o.makeHttpRequest(ApiUrl)
+		if err != nil {
+			return err
+		}
+		responseBody, err := ioutil.ReadAll(response.Body)
+		if err != nil {
+			return fmt.Errorf("failed to read response: %v", err)
+		}
+
+		err = getLogs(responseBody, &logList, o.Prefix)
+		if err != nil {
+			return err
+		}
+
+		err = printLogs(logList, streams, o.Limit)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (o *LogParameters) makeHttpRequest(baseUrl string) (*http.Response, error) {
+
+	var urlBuilder strings.Builder
+	urlBuilder.WriteString(baseUrl)
+	numParameters := 0
+	if len(o.Podname) > 0 {
+		urlBuilder.WriteString("?")
+		urlBuilder.WriteString("podname=" + o.Podname)
+		numParameters = numParameters + 1
+	}
+	if len(o.Namespace) > 0 {
+		if numParameters == 0 {
+			urlBuilder.WriteString("?")
+		} else {
+			urlBuilder.WriteString("&")
+		}
+		urlBuilder.WriteString("namespace=" + o.Namespace)
+		numParameters = numParameters + 1
+	}
+	if len(o.Limit) > 0 {
+		if numParameters == 0 {
+			urlBuilder.WriteString("?")
+		} else {
+			urlBuilder.WriteString("&")
+		}
+		urlBuilder.WriteString("maxlogs=" + o.Limit)
+		numParameters = numParameters + 1
+	}
+	if len(o.Level) > 0 {
+		if numParameters == 0 {
+			urlBuilder.WriteString("?")
+		} else {
+			urlBuilder.WriteString("&")
+		}
+		urlBuilder.WriteString("level=" + o.Level)
+		numParameters = numParameters + 1
+	}
+	if len(o.StartTime) > 0 && len(o.EndTime) > 0 {
+		if numParameters == 0 {
+			urlBuilder.WriteString("?")
+		} else {
+			urlBuilder.WriteString("&")
+		}
+		urlBuilder.WriteString("starttime=" + o.StartTime)
+		urlBuilder.WriteString("&finishtime=" + o.EndTime)
+		numParameters = numParameters + 1
+	}
+	response, err := http.Get(urlBuilder.String())
+
 	if err != nil {
-		return fmt.Errorf("Failed to read Response: %v", err)
+		return nil, fmt.Errorf("http request failed: %v", err)
 	}
+	return response, nil
+}
+
+func printLogs(logList []string, streams genericclioptions.IOStreams, limit string) error {
+
+	if len(logList) == 0 {
+		_, err := fmt.Fprintf(streams.Out, "No Logs Present, or logs from invalid resources were requested")
+		if err != nil {
+			return fmt.Errorf("an error occurred while printing logs: %v", err)
+		}
+		return nil
+	}
+
+	var maxLogs int
+	var err error
+
+	if len(limit) > 0 {
+		maxLogs, err = strconv.Atoi(limit)
+		if err != nil {
```

Comment:
While writing tests, I observed that it is not returning error for negative value of `limit`. How about changing this condition to `if err != nil || maxLogs < 0`?

Reply:
At the moment, if the server sends a `400` status code, or if logs are not present for the requested resources, the message displayed is `No Logs Present, or logs from invalid resources were requested`, which covers the case in which `limit` is negative. This could be changed to a more specific error message instead though.